packages/commonwealth/client/scripts/views/pages/CommunityManagement/Contests/ManageContest/steps/DetailsFormStep/DetailsFormStep.tsx

```diff
@@ -0,0 +1,443 @@
+import React, { useEffect, useState } from 'react';
+
+import { useCommonNavigate } from 'navigation/helpers';
+import app from 'state';
+import { useFetchTopicsQuery } from 'state/api/topics';
+import {
+  CWCoverImageUploader,
+  ImageBehavior,
+} from 'views/components/component_kit/cw_cover_image_uploader';
+import { CWDivider } from 'views/components/component_kit/cw_divider';
+import { CWText } from 'views/components/component_kit/cw_text';
+import { CWButton } from 'views/components/component_kit/new_designs/CWButton';
+import { CWForm } from 'views/components/component_kit/new_designs/CWForm';
+import { CWTextInput } from 'views/components/component_kit/new_designs/CWTextInput';
+import { MessageRow } from 'views/components/component_kit/new_designs/CWTextInput/MessageRow';
+import { CWRadioButton } from 'views/components/component_kit/new_designs/cw_radio_button';
+import { CWToggle } from 'views/components/component_kit/new_designs/cw_toggle';
+import CommunityManagementLayout from 'views/pages/CommunityManagement/common/CommunityManagementLayout';
+
+import {
+  ContestFeeType,
+  ContestFormSubmitValues,
+  ContestRecurringType,
+  LaunchContestStep,
+} from '../../types';
+import PayoutRow from './PayoutRow';
+import {
+  INITIAL_PERCENTAGE_VALUE,
+  MAX_WINNERS,
+  MIN_WINNERS,
+  initialPayoutStructure,
+  prizePercentageOptions,
+} from './utils';
+import { detailsFormValidationSchema } from './validation';
+
+import './DetailsFormStep.scss';
+
+interface DetailsFormStepProps {
+  contestId?: string;
+  onSetLaunchContestStep: (step: LaunchContestStep) => void;
+}
+
+const DetailsFormStep = ({
+  contestId,
+  onSetLaunchContestStep,
+}: DetailsFormStepProps) => {
+  const navigate = useCommonNavigate();
+
+  const [payoutStructure, setPayoutStructure] = useState<number[]>(
+    initialPayoutStructure,
+  );
+  const [prizePercentage, setPrizePercentage] = useState(
+    INITIAL_PERCENTAGE_VALUE,
+  );
+  const [toggledTopicList, setToggledTopicList] = useState<
+    {
+      name: string;
+      id: number;
+      checked: boolean;
+    }[]
+  >([]);
+  const [allTopicsToggled, setAllTopicsToggled] = useState(true);
+  const [isProcessingProfileImage, setIsProcessingProfileImage] =
+    useState(false);
+
+  const { data: topicsData } = useFetchTopicsQuery({
+    communityId: app.activeChainId(),
+  });
+
+  const editMode = !!contestId;
+  const payoutRowError = payoutStructure.some((payout) => payout < 1);
+  const topicsEnabledError = toggledTopicList.every(({ checked }) => !checked);
+  const totalPayoutPercentage = payoutStructure.reduce(
+    (acc, val) => acc + val,
+    0,
+  );
+  const totalPayoutPercentageError = totalPayoutPercentage !== 100;
+
+  const sortedTopics = [...topicsData]?.sort((a, b) => {
+    if (!a.order || !b.order) {
+      return 1;
+    }
+
+    // if order is not defined, push topic to the end of the list
+    return a.order - b.order;
+  });
```

Comment:
my opinion on memoization is that it should be used when there is a visible performance drop during rerender phase. Otherwise it just creates mess in the code. If there were thousands of topics then maybe, but here we talk about 10-20.